apps/app/src/components/ActiveLink.tsx

```diff
@@ -4,25 +4,27 @@ import Link from "next/link";
 import { usePathname } from "next/navigation";
 import { twMerge } from "tailwind-merge";
 import type { ComponentProps } from "react";
+import { useDevFAQRouter } from "../hooks/useDevFAQRouter";
 
 type ActiveLinkProps = Readonly<{
 	activeClassName: string;
 }> &
 	ComponentProps<typeof Link>;
 
-export const ActiveLink = ({
-	href,
-	className,
-	activeClassName,
-	children,
-	...rest
-}: ActiveLinkProps) => {
+export const ActiveLink = ({ href, className, activeClassName, ...props }: ActiveLinkProps) => {
 	const pathname = usePathname();
+	const { queryParams } = useDevFAQRouter();
+
 	const isActive = pathname?.startsWith(href.toString());
 
 	return (
-		<Link href={href} className={twMerge(className, isActive && activeClassName)} {...rest}>
-			{children}
-		</Link>
+		<Link
+			href={{
+				pathname: href.toString(),
+				query: queryParams,
```

Comment:
W większości przypadków nie chcemy kopiować obecnego query.
W zasadzie to chcemy to robić tylko przy przechodzeniu między stronami w paginacji i przy zmianie technologii, nie?

Reply:
Tak, jeszcze przy zmianie poziomu trudności.